test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -2207,3 +2207,77 @@ def test_measure_invalid_qubit():
 def test_measure_qubit_out_of_range(qasm, expected):
     with pytest.raises(IndexError, match=expected):
         Interpreter().build_circuit(qasm)
+
+
+@pytest.mark.parametrize(
+    "qasm, expected",
+    [
+        (
+            """
+                bit[2] b;
+                qubit["10"] r1;
+                b = measure r1;
+            """,
+            [0, 1],
+        ),
+        (
+            """
+                bit[3] b;
+                qubit["11"] r1;
+                b = measure r1;
+            """,
+            [0, 1, 2],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit[!"1"] r1;
+                b = measure r1;
+            """,
+            [],
+        ),
+        (
+            """
+                qubit["1" ^ "0"] r1;
+            """,
+            [],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit["1" != "0"] r1;
+                b = measure r1;
+            """,
+            [0],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit["1" == "0"] r1;
+                b = measure r1;
+            """,
+            [],
+        ),
+        (
+            """
+                bit[1] b;
+                qubit[1] r1;
+                h r1["0" << "1"];
+                b = measure r1;
+            """,
```

Comment:
[minor] these tests which run gates don't validate that they are run correctly, and they also only test `h r1[0]` and not any other qubit indices. Could they be expanded a bit - for example, to test running gates on additional qubits beyond 0?

Reply:
I believe this would be (a bit) better as `0 << X` is always `0`; using `1 << 1` gives `2`:
```
            """
                bit[3] b;
                qubit[3] r1;
                h r1["1" << "1"]; 
                b = measure r1;
            """,
 ```

I don't have a quantum background, but by me eye this should result in the probability state of `r1[2] = 0.5` and the other qubits are default.